datafusion/core/src/physical_plan/filter.rs

```diff
@@ -672,4 +675,309 @@ mod tests {
 
         Ok(())
     }
+
+    #[tokio::test]
+    async fn test_filter_statistics_multiple_columns() -> Result<()> {
```

Comment:
👍  thank you for these tests

Stylistically they might be easier to write if you could avoid the setup of StatisticsExec and FilterExec (and just make the input / predicate / output. However, that is just a minor suggestion